src/librustc_mir/transform/mod.rs

```diff
@@ -229,6 +231,8 @@ fn mir_const(tcx: TyCtxt<'_>, def_id: DefId) -> Steal<Body<'_>> {
         None,
         MirPhase::Const,
         &[&[
+            // MIR-level lints.
+            &check_packed_ref::CheckPackedRef,
```

Comment:
Looks like an alternative to this would be to add the lint here:

https://github.com/rust-lang/rust/blob/31add7e60709445617ab54a69f6f21cfcb2e3122/src/librustc_mir_build/build/mod.rs#L184